model/PermissionProvider.php

```diff
@@ -63,29 +67,67 @@ public function getPermissions(User $user, array $resourceIds) {
         return $rights;
     }
 
-    protected function getStateObjects($stateIds) {
+    protected function getStateObjects($stateIds)
+    {
         $wfmodel = $this->getServiceManager()->get(WorkflowModel::SERVICE_ID);
         $stateCache = array();
         foreach ($stateIds as $stateId) {
             if (!empty($stateId)) {
-                $stateCache[$stateId] = $wfmodel->getState($stateId);
+                $state = $wfmodel->getState($stateId);
+                $stateWithRoles = $this->addAtomicRoles($state);
+                $stateCache[$stateId] = $stateWithRoles;
             }
         }
         return $stateCache;
     }
 
     /**
      * (non-PHPdoc)
+     *
      * @see \oat\generis\model\data\PermissionInterface::onResourceCreated()
      */
-    public function onResourceCreated(\core_kernel_classes_Resource $resource) {
+    public function onResourceCreated(\core_kernel_classes_Resource $resource)
+    {
     }
 
     /**
      * (non-PHPdoc)
+     *
      * @see \oat\generis\model\data\permission\PermissionInterface::getSupportedRights()
      */
-    public function getSupportedRights() {
+    public function getSupportedRights()
```

Comment:
Just a suggestion, for methods from interface, we can just omit all annotations block (remove it) and it will automatically inherit from interface.